src/compiler/transformers/classFields.ts

```diff
@@ -1341,6 +1544,15 @@ namespace ts {
         }
     }
 
+    function createPrivateStaticFieldInitializer(variableName: Identifier, initializer: Expression | undefined) {
+        return factory.createAssignment(
+            variableName,
+            factory.createObjectLiteralExpression([

```

Comment:
Since we use a function to set the value of the static property and JS can't pass by reference, this was the simplest solution I could think of: Pass in the wrapper object to the static fields set function in order to allow it to mutate the value property.

Reply:
As we talked offline, we will stick with the container object.